pkg/streamingpromql/engine_fuzz_test.go

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package streamingpromql
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/promqltest"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/querier/stats"
+	"github.com/grafana/mimir/pkg/streamingpromql/testutils"
+)
+
+const (
+	// commentLinePrefix is the prefix we use in the query seeding file to indicate a comment
+	commentLinePrefix = "#"
+	// ignoreAnnotationDirective is the prefix we use in the query seeding to flag that we will not match on query result annotations - trailing space is intentional
+	ignoreAnnotationDirective = "[ignore_annotation] "
+)
+
+// A queryEngines holds our Prometheus and Mimir QueryEngines and a Queryable (storage).
+// A queryEngines also holds rules for how we process certain queries.
+type queryEngines struct {
```

Comment:
[nit] Perhaps `fuzzConfig` or `fuzzTestEnvironment` would be clearer names given this struct holds more than just some engines? Then we could change the name of `buildQueryEngines` to match, given it does more than just create query engines.